### PR TITLE
Upgrade sf sdk version 1.0.0, add support for cert by common name and add tests

### PR DESCRIPTION
**Pull request by a-santamaria**
## Description

-Updated service fabric sdk version to 1.0.0
-Added optional parameters -CertCommonName and -CertIssuerThumbprint to New-AzureRmServiceFabricCluster and Add-AzureRmServiceFabricClusterCertificate to support certificates by common names
-Added the following tests:
--TestNewAzureRmServiceFabricClusterCNCert
--TestAddAzureRmServiceFabricApplicationCertificate
--TestAddAzureRmServiceFabricClusterCertificateCNNotAllowed
--TestAddAzureRmServiceFabricClusterCertificateCNNotAllowed
--TestRemoveAzureRmServiceFabricClusterCertificateNotAllowed
-Added priority to the tests because we need to record tests in a particular order as we use one Service Fabric cluster configured before the tests start

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only) Not yet, here I've created the [reviewIssue](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/204)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist

**Comment by a-santamaria** [comment]
@maddieclayton I'm getting StaticAnalysis failure on the build in the Breaking changes file I see these 3:

The property   'SerivceTypeHealthPolicies' of type   'Microsoft.Azure.Management.ServiceFabric.Models.ApplicationHealthPolicy' has   been removed.

The property 'ApplicationHealthPolicies'   of type   'Microsoft.Azure.Management.ServiceFabric.Models.ClusterUpgradeDeltaHealthPolicy'   has been removed.

The property 'OverrideUserUpgradePolicy'   of type   'Microsoft.Azure.Management.ServiceFabric.Models.ClusterUpgradePolicy' has   been removed.

I don't understand why these are breaking changes. What should I do for this?

**Comment by maddieclayton** [comment]
@a-santamaria Please remove the deprecation messages in this PR (they will be released in this release, but should be removed for next release), and follow the instructions in the error message of the build to suppress the breaking change errors.

**Comment by a-santamaria** [comment]
thanks @maddieclayton I removed the deprecation messages and suppress the breaking change errors

**Comment by cormacpayne** [comment]
@a-santamaria I re-targeted this branch to the new default branch, `master`, so you will need to pull from that branch to resolve the merge conflicts. 

_Note_: since the next release is the GA of the `Az` module, we have removed all reference docs to `AzureRM` cmdlets, so you will need to regenerate your help based on the `Az` cmdlets.

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/NewAzureRmServiceFabricCluster.cs [hunk 1]
```diff
@@ -176,6 +176,21 @@ public override string KeyVaultResouceGroupLocation
         [ValidateNotNullOrEmpty]
         public override string KeyVaultName { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ByExistingKeyVault,
```
Comment (maddieclayton) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/NewAzureRmServiceFabricCluster.cs]: Please remove the ValueFromPipeline from both parameters.

**Comment by a-santamaria** [comment]
thanks @cormacpayne I fix the conflicts and regenerated the help files. @maddieclayton could you take a look? thanks

File: src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs [hunk 2]
```diff
@@ -127,9 +127,28 @@ public virtual string CertificateSubjectName
...
                     this.certificateSubjectName = string.Concat("cn=", value);
                 }
+                else
+                {
+                    this.certificateSubjectName = value;
+                }
             }
         }
 
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ParameterSetName = ByExistingKeyVault,
```
Comment (maddieclayton) [inline src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs]: Remove valuefrompipeline from these two parameters as well. It doesn't make sense for a string parameter to have this attribute.

Comment (a-santamaria) [reply src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCertificateCmdlet.cs]: done, thanks

**Pull request merged**